### ISSUE TITLE
Getting command line via Diagnostic IPC APIs doesn't work properly in .NET 6

### ISSUE THREAD

**Issue by valco1994** [dotnet/runtime#64270]
### Description

I want to get a command line for the process being profiled via EventPipe on Linux and macOS. I would like to get it as fast as possible.

There are several expected ways to get a command line using Diagnostic IPC:

- run EventPipe session and listen to `RuntimeInformationStart` event from `Microsoft-Windows-DotNETRuntime` or `ProcessInfo` event from `Microsoft-DotNETCore-EventPipe`
- use `ProcessInfo` or `ProcessInfo2` command

`Microsoft.Diagnostics.NETCore.Client.DiagnosticsClient` has the `GetProcessInfo` method, but it's declared as internal, so I can't use it for issue demonstrating. I used my own implementation to communicate with Diagnostic IPC APIs.

Actually, I would expect that .NET runtime originally has proper value for its own process command line and provide correct value in the aforementioned events. In .NET 5, it was not completely true, especially for profiling with DiagnosticPorts (aka reversed EventPipe) and using several EventPipe sessions in parallel. But it was possible to work around this issue by polling the profiled process with the `ProcessInfo` command. And soon enough command line occurred to be initialized properly.

But as far as I can see, in .NET 6 it occurred to be broken completely. `ProcessInfo` command result contains only an executable name, not a full command line. In some rare cases, it is possible to get the proper command line at the end of application work, but usually, it's not provided. Getting command line from events works only in the case of the only session and is sent at the end of application work. In the case of several sessions, the actual command line is not provided at all.

### Reproduction Steps

Try to poll  .NET 6 process command line using the `ProcessInfo` command

### Expected behavior

The actual command line is provided. In the perfect case as the answer to the first request.

### Actual behavior

Usually, the actual command line is not received at all

### Regression?

Yes, it was not perfect but much better in .NET 5

### Known Workarounds

As I have described, there is a workaround for .NET 5, but I don't know a workaround for .NET 6

### Configuration

.NET 6
macOS Big Sur (both x64 and aarch64), Linux Ubuntu 20.04 x64 (most likely it's the same on Linux aarch64, but I didn't check)

**Comment by jander-msft** [comment]
This is a dup of #59439

**Comment by valco1994** [comment]
Thank you, @jander-msft, I missed this issue. But actually, they are strongly related, but not full duplicates. Issue #59439 is only about getting the command line during suspension on the process start with Diagnostic Ports. And I can see the same on both .NET 5 and .NET 6. But in this issue, I report that in .NET 6, it occurred to be impossible to get a proper command line during the whole process lifetime, while I expect that it should be initialized at least a short time after runtime resuming (and it worked like that in .NET 5).

**Comment by jander-msft** [comment]
Issue #59439 goes on to describe the behavior after the runtime is resumed. The only thing that isn't explicitly captured in the issue is that the result of the first invocation of the ProcessInfo command is cached for the lifetime of the process on non-Windows platforms. So depending on when the first invocation occurs, you may only get the process name for the lifetime of the process or you may get the full command line.

The caching behavior has already been fixed in 7.0 with #60275
Tests to validate and prevent regression are in #63382
Backport to 6.0 is in draft in #63356

**Comment by valco1994** [comment]
Oh, thank you, it's good news!

It would be very very nice to have it in .NET 6. Could you tell me if I can vote somehow for this fix to be included in .NET 6?

**Comment by tommcdon** [comment]
> Oh, thank you, it's good news!
> 
> It would be very very nice to have it in .NET 6. Could you tell me if I can vote somehow for this fix to be included in .NET 6?

Thanks for the feedback.  Your comment on this PR counts as a vote towards backporting to 6.0.  The work to backport the PR is on the backlog for now.  Is this something critical to fix right now for your team?  Otherwise we plan to revisit the backport in the next 2-3 months.

**Comment by valco1994** [comment]
It's not critical but quite annoying. And considering that
-  the risk of backporting is said to be minimal
- .NET 6 is an LTS version
- it is a regression introduced in .NET 6

I suppose that it's worth being backported. And to be honest, I suppose that it will be postponed, it could never be backported at all 😞

**Comment by tommcdon** [comment]
@valco1994 Thanks for the feedback.  This is on our backlog for a .NET 6 backport.  We hope to get to this issue in the next 2-3 months.

**Comment by tommcdon** [comment]
Fixed in 6.0 https://github.com/dotnet/runtime/pull/63356
